### PR TITLE
feat(`claim`): implement `CompleteMission` 

### PR DESCRIPTION
**Pull request by lubtd**
- Implement `CompleteMission` that defines logic for claimable airdrop distribution
- Add method tests
- Add airdrop supply initialization (module account mint the airdrop supply at genesis initialization)

### CODE REVIEW

File: x/claim/keeper/airdrop_supply.go [hunk 1]
```diff
@@ -26,3 +27,34 @@ func (k Keeper) GetAirdropSupply(ctx sdk.Context) (val sdk.Coin, found bool) {
...
+func (k Keeper) InitializeAirdropSupply(ctx sdk.Context, airdropSupply sdk.Coin) error {
+	// get the eventual existing balance of the module for the airdrop supply
+	moduleBalance := k.bankKeeper.GetBalance(
+		ctx,
+		k.accountKeeper.GetModuleAddress(types.ModuleName),
+		airdropSupply.Denom,
+	)
+
+	// if the module has an existing balance, we burn the entire balance
+	if moduleBalance.IsPositive() {
```
Comment (aljo242) [inline x/claim/keeper/airdrop_supply.go:47]: What is the scenario where this will happen?

File: x/claim/keeper/claim_record.go [hunk 2]
```diff
@@ -40,9 +35,7 @@ func (k Keeper) RemoveClaimRecord(
 	index string,
```
Comment (aljo242) [inline x/claim/keeper/claim_record.go:35]:
```suggestion
	address string,
```

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: x/claim/keeper/airdrop_supply.go (hunk 1 above)
Comment (lubtd) [reply x/claim/keeper/airdrop_supply.go:47]: In the case you export and import genesis

File: x/claim/genesis_test.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 
 	"github.com/stretchr/testify/require"
```
Comment (Pantani) [inline x/claim/genesis_test.go:8]:
```suggestion
```

File: x/claim/keeper/grpc_query_airdrop_supply_test.go [hunk 4]
```diff
@@ -1,22 +1,26 @@
 package keeper_test
 
 import (
+	"github.com/tendermint/spn/testutil/sample"
```
Comment (Pantani) [inline x/claim/keeper/grpc_query_airdrop_supply_test.go:4]: wrong import sort, runs `make format`

File: x/claim/keeper/grpc_query_airdrop_supply_test.go [hunk 5]
```diff
@@ -1,22 +1,26 @@
...
 	"github.com/tendermint/spn/x/claim/types"
 )
 
 func TestAirdropSupplyQuery(t *testing.T) {
-	keeper, ctx := keepertest.ClaimKeeper(t)
+	ctx, tk, _ := testkeeper.NewTestSetup(t)
+
 	wctx := sdk.WrapSDKContext(ctx)
-	item := createTestAirdropSupply(keeper, ctx)
+	sampleSupply := sample.Coin(r)
```
Comment (Pantani) [inline x/claim/keeper/grpc_query_airdrop_supply_test.go:21]:
```suggestion

	var (
		ctx, tk, _	= testkeeper.NewTestSetup(t)
		wctx 		= sdk.WrapSDKContext(ctx)
		sampleSupply	= sample.Coin(r)
	)
```

File: x/claim/keeper/grpc_query_airdrop_supply_test.go [hunk 6]
```diff
@@ -1,22 +1,26 @@
...
+	tk.ClaimKeeper.SetAirdropSupply(ctx, sampleSupply)
+
 	for _, tc := range []struct {
 		desc     string
 		request  *types.QueryGetAirdropSupplyRequest
```
Comment (Pantani) [inline x/claim/keeper/grpc_query_airdrop_supply_test.go:26]: We can compare only the `AirdropSupply`

```suggestion
		request  types.AirdropSupply
```

File: x/claim/keeper/grpc_query_claim_record_test.go [hunk 7]
```diff
@@ -67,9 +68,10 @@ func TestClaimRecordQuerySingle(t *testing.T) {
 }
 
 func TestClaimRecordQueryPaginated(t *testing.T) {
-	keeper, ctx := keepertest.ClaimKeeper(t)
+	ctx, tk, _ := testkeeper.NewTestSetup(t)
+
 	wctx := sdk.WrapSDKContext(ctx)
-	msgs := createNClaimRecord(keeper, ctx, 5)
+	msgs := createNClaimRecord(tk.ClaimKeeper, ctx, 5)
```
Comment (Pantani) [inline x/claim/keeper/grpc_query_claim_record_test.go:74]:
```suggestion
	var (
		ctx, tk, _	= testkeeper.NewTestSetup(t)
		wctx		= sdk.WrapSDKContext(ctx)
		msgs		= createNClaimRecord(tk.ClaimKeeper, ctx, 5)
	)
```

File: x/claim/keeper/mission.go [hunk 8]
```diff
@@ -6,9 +6,18 @@ import (
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
```
Comment (Pantani) [inline x/claim/keeper/mission.go:10]:
```suggestion
	"github.com/cosmos/cosmos-sdk/store/prefix"
	sdk "github.com/cosmos/cosmos-sdk/types"
	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
	
	spnerrors "github.com/tendermint/spn/pkg/errors"
```

File: x/claim/keeper/grpc_query_airdrop_supply_test.go (hunk 6 above)
Comment (lubtd) [reply x/claim/keeper/grpc_query_airdrop_supply_test.go:26]: I don't get your comment

The method is request -> response, therefore we assert the expected response with the actual one
There is one test case with `nil` request

**Review by aljo242 — DISMISSED** [review]

File: x/claim/keeper/grpc_query_airdrop_supply_test.go (hunk 6 above)
Comment (Pantani) [reply x/claim/keeper/grpc_query_airdrop_supply_test.go:27]: yes, but we don't need the entire object to compare. The test case can be simpler

```go
		{
			desc:     "First",
			request:  &types.QueryGetAirdropSupplyRequest{},
			supplyResponse: sampleSupply,
		},
```

Comment (Pantani) [reply x/claim/keeper/grpc_query_airdrop_supply_test.go:27]: and we can compare only the supply

```go
				require.Equal(t,
					nullify.Fill(tc.supplyResponse),
					nullify.Fill(response. AirdropSupply),
				)
```

Comment (lubtd) [reply x/claim/keeper/grpc_query_airdrop_supply_test.go:27]: I see.
I find it more visual to use response since it's the format we have for all grpc tests

**Review by Pantani — DISMISSED** [review]

**Pull request merged**